ebu_tt_live/bindings/test/test_role_mixin.py

```diff
@@ -0,0 +1,86 @@
+from unittest import TestCase
+
+from ebu_tt_live.bindings import d_div_type, d_br_type, d_body_type, d_p_type, d_span_type, div_type, \
+    br_type, body_type, p_type, span_type
+
+from ebu_tt_live.bindings._ebuttm import pMetadata_type, divMetadata_type
+
+
+class TestElementRoleMixin(TestCase):
+    class Metadata:
+        def __init__(self, role=None):
+            self.role = role
+
+    def setUp(self):
+        self.elements = {
+            'd_div': d_div_type(),
+            'd_br': d_br_type(),
+            'd_body': d_body_type(),
+            'd_p': d_p_type(),
+            'd_span': d_span_type(),
+            'div': div_type(),
+            'br': br_type(),
+            'body': body_type(),
+            'p': p_type(),
+            'span': span_type()
+        }
+        self.dataset = {'role_stack': []}
+        self.metadata_types = {
+            'div': divMetadata_type,
+            'p': pMetadata_type
+        }
+
+    def test_initial_computed_role(self):
+        """Test that computed_role is None when no role is set and role_stack is empty for different element types."""
+        for element_name, element in self.elements.items():
+            with self.subTest(element=element_name):
+                self.assertEqual(element.computed_roles, set(),
+                                 f"Initial computed role should be None for {element_name}.")
+
+    def test_role_propagation(self):
+        """Test proper propagation of directly set role to computed_role for different element types."""
```

Comment:
Calling this "propagation" gives the idea that it's about role values being propagated between elements. Maybe call if `test_directly_specified_role` or something like that?

Reply:
updating